sunkit_spex/models/physical/thermal.py

```
@@ -719,15 +1262,33 @@ def _weight_emission_bins(
     return new_line_intensities, neighbor_intensities, neighbor_iline
 
 
-def _sanitize_inputs(energy_edges, temperature):
-    # Convert inputs to known units and confirm they are within range.
+def _sanitize_inputs(energy_edges, temperature, emission_measure, observer_distance):
+
     if energy_edges.isscalar or len(energy_edges) < 2 or energy_edges.ndim > 1:
         raise ValueError("energy_edges must be a 1-D astropy Quantity with length greater than 1.")
-    energy_edges_keV = energy_edges.to_value(u.keV)
-    temperature_K = temperature.to_value(u.K)
+
+    if isinstance(energy_edges, Quantity):
+        energy_edges_keV = energy_edges.to(u.keV)
+    else:
+        energy_edges_keV = energy_edges
```

I'm wondering if this is the behaviour we want. We want the values to be in keV regardless of the input quantity so should we not always want to cast it (e.g., `<< u.keV`). I'm mainly concerned with the line that leaves the input alone if the input isn't a `Quantity`.

---

As with the `if/else` statement discussed above, it is essential for the fitting that these functions work with no units, perhaps we could throw an error in the initialise step if the energy_edges are not in kev, or a compatible unit?